### ISSUE TITLE
IP errors and connection to other dimensions

### ISSUE THREAD

**Issue by Dragons1107** [popstarfreas/Dimensions#119]
dimensions does not work correctly for me since when I place the command to join another dimension I get shifting dimension and nothing happens, apart from when I ban a user or see the console all the user ips are changed by the ip of the server.

**Comment by popstarfreas** [comment]
Did you check your TShock server logs to see if there are any errors on the dimension you are trying to switch to?

**Comment by Dragons1107** [comment]
> Did you check your TShock server logs to see if there are any errors on the dimension you are trying to switch to?

I checked the action of the consoles of each server when I tried to join the other dimension and what appears in the console of the dimension that I want to join when placing the command is that I don't have the same version

**Comment by popstarfreas** [comment]
What version of dimensions are you using?
What Terraria version is your server?

**Comment by Dragons1107** [comment]
> 1. What version of dimensions are you using?
> 2. What Terraria version is your server?

1. The version of dimensions that I use is v5.9.1
2. The version of Terraria that I use is 1.4.4.9

**Comment by popstarfreas** [comment]
This is the latest version of Dimensions: https://github.com/popstarfreas/Dimensions/releases/tag/v5.10.0-beta.1

See if that makes a difference.

**Comment by Dragons1107** [comment]
> This is the latest version of Dimensions: https://github.com/popstarfreas/Dimensions/releases/tag/v5.10.0-beta.1
> See if that makes a difference.

Now the dimensions work, but the ip problem continues, when I ban a user I ban the server's ip and they all get that they are banned

**Comment by Dragons1107** [comment]
> This is the latest version of Dimensions: https://github.com/popstarfreas/Dimensions/releases/tag/v5.10.0-beta.1
> 
> See if that makes a difference.

Now it does teleport to the other dimensions but the problem that when banning a player and that all of them say they are banned persists and there are 2 new problems, there are players who appear nameless and invisible.

**Comment by popstarfreas** [comment]
>  when banning a player and that all of them say they are banned persists

What do you mean by this?

>  there are players who appear nameless and invisible.

This has been mentioned before I think: https://github.com/popstarfreas/Dimensions/issues/110 for now I'm not able to replicate it easily and I'm not yet running the latest version, so until then I won't be able to properly fix it.

**Comment by Dragons1107** [comment]
> > al banear a un jugador y que todos digan que están baneados persiste
> 
> ¿Qué quiere decir con esto?
When I ban a player, it doesn't ban their ip, it bans the server's ip and that causes all the players to be banned, when I activate dimensions all the players' ips are replaced by the server's ip

**Comment by Dragons1107** [comment]
> Esto se ha mencionado antes, creo: [# 110](https://github.com/popstarfreas/Dimensions/issues/110) por ahora no puedo replicarlo fácilmente y aún no estoy ejecutando la última versión, por lo que hasta entonces no podré solucionarlo correctamente.

Yes, that's exactly what happens

**Comment by popstarfreas** [comment]
> > al banear a un jugador y que todos digan que están baneados persiste
> > ¿Qué quiere decir con esto?
> 
> When I ban a player, it doesn't ban their ip, it bans the server's ip and that causes all the players to be banned, when I activate dimensions all the players' ips are replaced by the server's ip

That sounds like you didn't do step 2:
https://github.com/popstarfreas/Dimensions#step-2-install-the-plugin

The plugin is what overwrites the IP to be correct

**Comment by Dragons1107** [comment]
> > > al banear a un jugador y que todos digan que están baneados persiste
> > > ¿Qué quiere decir con esto?
> > 
> > 
> > When I ban a player, it doesn't ban their ip, it bans the server's ip and that causes all the players to be banned, when I activate dimensions all the players' ips are replaced by the server's ip
> 
> That sounds like you didn't do step 2: https://github.com/popstarfreas/Dimensions#step-2-install-the-plugin
> 
> The plugin is what overwrites the IP to be correct

the plugin is installed, is it necessary to activate the GeoIP option that appears in the .json?

**Comment by popstarfreas** [comment]
No, GeoIP is if you use the GeoIP tshock option. You should be able to check in the tshock logs or serverlog.txt to see if the plugin is beinged loaded.

**Comment by Dragons1107** [comment]
> No, GeoIP is if you use the GeoIP tshock option. You should be able to check in the tshock logs or serverlog.txt to see if the plugin is being loaded.
Although the plugin loads, the ip error continues

**Comment by popstarfreas** [comment]
The server I use this for (Dark Gaming) is currently in the processing of updating all our dimensions to 1.4.4.9. I'll check and see if we had to update the plugin for it to work or not, and if so, I'll make a new release with an updated version.

**Comment by Dragons1107** [comment]
> The server I use this for (Dark Gaming) is currently in the processing of updating all our dimensions to 1.4.4.9. I'll check and see if we had to update the plugin for it to work or not, and if so, I'll make a new release with an updated version.

Ok, so I'll wait for the next release to update dimensions

**Comment by popstarfreas** [comment]
This is the one we are using on Dark Gaming: https://github.com/popstarfreas/Dimensions-TerrariaServer/releases/tag/v1.5.0

Maybe try that and see?

**Comment by Dragons1107** [comment]
> 

It works but still has errors, invisible people, names are not seen, the map seems to merge with the dimensions.

**Comment by popstarfreas** [comment]
When you say map do you mean the minimap? That's just how the game itself works, it doesn't have the capability to have a minimap per dimension.

The other things you mention are likely to be fixed soon as my server updates to the latest version.

**Comment by Dragons1107** [comment]
> When you say map do you mean the minimap? That's just how the game itself works, it doesn't have the capability to have a minimap per dimension.
> 
> The other things you mention are likely to be fixed soon as my server updates to the latest version.

Well that's the least of it, and is there a way to solve the other errors?

**Comment by popstarfreas** [comment]
Depends specifically on which errors.

I am right now in the process of deploying the latest Dimensions at Dark Gaming, with the help of users we will be testing and fixing issues to get it to be stable for deployment.

The only thing we don't specifically have in our setup is multiple SSC servers, there were previously (years ago) issues created around switching directly between SSC servers that share the same inventory, which we do not have to test.

**Comment by popstarfreas** [comment]
Please note that a new release has some fixes for issues:
https://github.com/popstarfreas/Dimensions/releases/tag/v6.0.0-beta.0

**Comment by popstarfreas** [comment]
Are you still having this issue?
